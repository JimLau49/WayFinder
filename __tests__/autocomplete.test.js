import React from "react";
import { TouchableOpacity } from "react-native";
import AutoComplete from "../src/components/search/autocomplete.component";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

let mockPOIs = [
  {
    id: "c4541028-739d-4651-b9f5-dc4d54054807",
    displayName: "Stairs 2",
    description: "",
    location: {
      latitude: 45.497482,
      longitude: -73.579034
    },
    buildingId: "BuildingId.H",
    level: 8,
    category: "Stairs"
  },
  {
    id: "f5a7d95c-170b-48c9-aac3-e9faafa6033d",
    displayName: "Stairs 3",
    description: "",
    location: {
      latitude: 45.497253,
      longitude: -73.579251
    },
    buildingId: "BuildingId.H",
    level: 8,
    category: "Stairs"
  }
];

describe("AutoComplete component", () => {
  it("should match snapshot", () => {
    const tree = renderer.create(<AutoComplete />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should run provided function on press", () => {
    const mockSetLocation = jest.fn();

    const wrapper = shallow(
      <AutoComplete
        autoCompleteValues={mockPOIs}
        setLocation={mockSetLocation}
      />
    );
    const touchableList = wrapper
      .find("FlatList")
      .props()
      .renderItem({ item: { displayName: "test" } });
    expect(touchableList.type).toBe(TouchableOpacity);
  });
});
