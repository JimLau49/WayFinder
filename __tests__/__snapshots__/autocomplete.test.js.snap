// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoComplete component should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F7F7FA",
        "borderColor": "#F7F7FA",
        "borderWidth": 2,
        "flex": 1,
        "position": "absolute",
        "top": 48,
        "width": 720,
        "zIndex": 3,
      },
      Object {},
    ]
  }
>
  <RCTScrollView
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    keyboardShouldPersistTaps="handled"
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="autoCompleteFlatList"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
